pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -161,13 +170,33 @@ type discovery struct {
 	// lastCPUTimeUpdate is the last time lastGlobalCPUTime was updated.
 	lastCPUTimeUpdate time.Time
 
+	lastNetworkStatsUpdate time.Time
+
 	containerProvider proccontainers.ContainerProvider
 	timeProvider      timeProvider
+	network           networkCollector
 }
 
-func newDiscovery(containerProvider proccontainers.ContainerProvider, tp timeProvider) *discovery {
+type networkCollectorCreator func(cfg *discoveryConfig) (networkCollector, error)
```

Comment:
Nit: I would have called it `networkCollectorFactory`.

Reply:
Fixed.